src/components/VideoCard/VideoCard.js

```diff
@@ -24,7 +37,20 @@ export const VideoCard = ({ video }) => {
               <p className="card-text">{video.channel}</p>
             </div>
 
-            <i>{<MdWatchLater size={20} />}</i>
+            <i
+              onClick={() =>
+                authState.token
+                  ? toggleWatchLater(
+                      video,
+                      dispatchData,
+                      authState.token,
+                      isWatchLater
+                    )
+                  : navigate("/login")
+              }
```

Comment:
Click Handler can be moved into a separate function

Reply:
Made the change....thanks!!